janusgraph-solr/src/main/java/org/janusgraph/diskstorage/solr/SolrIndex.java

```diff
@@ -937,8 +937,8 @@ private static void waitForRecoveriesToFinish(CloudSolrClient server, String col
                for (Map.Entry<String, Slice> entry : slices.entrySet()) {
                     Map<String, Replica> shards = entry.getValue().getReplicasMap();
                     for (Map.Entry<String, Replica> shard : shards.entrySet()) {
-                        String state = shard.getValue().getStr(ZkStateReader.STATE_PROP);
-                        if ((state.equals(Replica.State.RECOVERING) || state.equals(Replica.State.DOWN))
+                        String state = shard.getValue().getStr(ZkStateReader.STATE_PROP).toUpperCase();
+                        if ((state.equals(Replica.State.RECOVERING.name()) || state.equals(Replica.State.DOWN.name()))
```

Comment:
Can you clarify? I don't want to assume the state has a constant but I could do something like below though it's a little more code. Is this what you had in mind?

```java
final String stateStr = shard.getValue().getStr(ZkStateReader.STATE_PROP).toUpperCase();
try {
    final Replica.State state = Replica.State.valueOf(stateStr);
    if ((state.equals(Replica.State.RECOVERING) || state.equals(Replica.State.DOWN))
        && clusterState.liveNodesContain(shard.getValue().getStr(
        ZkStateReader.NODE_NAME_PROP))) {
        sawLiveRecovering = true;
    }
} catch (IllegalArgumentException e) { }
```